src/View/Widget/InputgroupTrait.php

```diff
@@ -0,0 +1,102 @@
+<?php
+namespace BootstrapUI\View\Widget;
+
+use BootstrapUI\View\Helper\OptionsAwareTrait;
+use Cake\View\Form\ContextInterface;
+
+trait InputgroupTrait
+{
+    use OptionsAwareTrait;
+
+    /**
+     * Render a text widget or other simple widget like email/tel/number.
+     *
+     * This method accepts a number of keys:
+     *
+     * - `name` The name attribute.
+     * - `val` The value attribute.
+     * - `escape` Set to false to disable escaping on all attributes.
+     * - `append` Append addon to input.
+     * - `prepend` Prepend addon to input.
+     *
+     * Any other keys provided in $data will be converted into HTML attributes.
+     *
+     * @param array $data The data to build an input with.
+     * @param \Cake\View\Form\ContextInterface $context The current form context.
+     * @return string
+     */
+    public function output(array $data, ContextInterface $context)
+    {
+        $data += [
+            'type' => 'text',
```

Comment:
This will incorrectly add `'type'` attribute to the `textarea` tag. It should be removed and instead set only in the `BasicWidget` before this method is called.